api/envoy/extensions/common/dynamic_forward_proxy/v3/dns_cache.proto

```diff
@@ -136,8 +136,9 @@ message DnsCacheConfig {
   // The timeout used for DNS queries. This timeout is independent of any timeout and retry policy
   // used by the underlying DNS implementation (e.g., c-areas and Apple DNS) which are opaque.
   // Setting this timeout will ensure that queries succeed or fail within the specified time frame
-  // and are then retried using the standard refresh rates. Defaults to 5s if not set.
-  google.protobuf.Duration dns_query_timeout = 11 [(validate.rules).duration = {gt {}}];
+  // and are then retried using the standard refresh rates. Setting it to 0 will disable the DNS
```

Comment:
I think the comment may be a bit ambiguous.
Is disabling (setting to 0) causes the timeout threshold to be infinite (most likely), or is it causing it to immediately fail and retry on the next event-loop?

FWIW, I guess that even without this change, one can set the timeout to a very large number, which will practically be infinite timeout threshold.

Reply:
There's one subtle difference. With creating an infinite timeout, we are still calling `enableTimer()` on the [DNS query timer](https://github.com/envoyproxy/envoy/blob/6fea4d7715f2c3e93dff1db371b105b9d494525c/source/extensions/common/dynamic_forward_proxy/dns_cache_impl.cc#L50_) whereas with this PR, the `enableTimer()` is never called ( (no `event_add` call).

This is why the comment talks about disabling the DNS query timeout. Having a DNS query timeout to 0 to mean it always fails the DNS query is not a feature that's very useful IMO.

Reply:
I agree that the latter (failing immediately) makes less sense. FWIW, the context to this was a comment (on code not API) that was added in https://github.com/envoyproxy/envoy/pull/37709/files#diff-0ef84bd563523122c7562eac53c5f0dd54954064920c6f8b2d207408d97c39ecR1046-R1048, which isn't directly related to your change, but it is interesting that others may have a different view on the benefits of setting the value to 0 in order to fail immediately :)

I suggest clarifying the comment, just to avoid the ambiguity (i.e., equals to infinite timeout).

Reply:
I updated the comment to mention about disabling Envoy DNS query timeout and using the underlying DNS implementation timeout since that's essentially what we want to achieve. IMHO, mentioning about 0 == infinite can cause more confusion given the code doesn't even set the DNS query timer when the value is 0, which is different than when the value is very large. LMK your thoughts.

Reply:
IMHO it shouldn't mention the retry-timer. I think that the current description is much better.